addons/binding/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/handler/HyperionHandler.java

```diff
@@ -0,0 +1,315 @@
+                .findFirst();
+        if (color.isPresent()) {
+            List<Integer> rgbVals = color.get().getRGBValue();
+            int r = rgbVals.get(0);
+            int g = rgbVals.get(1);
+            int b = rgbVals.get(2);
+            HSBType hsbType = HSBType.fromRGB(r, g, b);
+            updateState(CHANNEL_COLOR, hsbType);
+        } else {
+            updateState(CHANNEL_COLOR, UnDefType.NULL);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Hyperion thing handler.");
+        try {
+            Configuration config = thing.getConfiguration();
+            address = (String) config.get(PROP_HOST);
+            port = ((BigDecimal) config.get(PROP_PORT)).intValue();
+            refreshInterval = ((BigDecimal) config.get(PROP_POLL_FREQUENCY)).intValue();
+            priority = ((BigDecimal) config.get(PROP_PRIORITY)).intValue();
+
+            connection = new JsonTcpConnection(address, port);
+            connectFuture = scheduler.scheduleWithFixedDelay(connectionJob, 0, refreshInterval, TimeUnit.SECONDS);
+            refreshFuture = scheduler.scheduleWithFixedDelay(refreshJob, 0, refreshInterval, TimeUnit.SECONDS);
+        } catch (UnknownHostException e) {
+            logger.warn("Could not resolve host: {}", e.getMessage());
+            updateOnlineStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing of Hyperion thing handler.");
+        if (refreshFuture != null) {
+            refreshFuture.cancel(true);
+        }
+        if (connectFuture != null) {
+            connectFuture.cancel(true);
+        }
+        if (connection != null && connection.isConnected()) {
+            try {
+                connection.close();
+            } catch (IOException e) {
+                // do nothing
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command instanceof RefreshType) {
+                if (refreshFuture.isDone()) {
+                    refreshFuture = scheduler.scheduleWithFixedDelay(refreshJob, 0, refreshInterval, TimeUnit.SECONDS);
+                } else {
+                    logger.debug("Previous refresh not yet completed");
+                }
+            } else if (CHANNEL_BRIGHTNESS.equals(channelUID.getId())) {
+                handleBrightness(command);
+            } else if (CHANNEL_COLOR.equals(channelUID.getId())) {
+                handleColor(command);
+            } else if (CHANNEL_CLEAR.equals(channelUID.getId())) {
+                handleClear(command);
+            } else if (CHANNEL_EFFECT.equals(channelUID.getId())) {
+                handleEffect(command);
+            }
+        } catch (IOException e) {
+            logger.warn("Unable to send command: {}", command.toString());
```

Comment:
reduce to debug and remove `.toString()` (everywhere where you have this in logging parameters - slf4j automatically calls a `toString()` on any object).

Reply:
done